### PR TITLE
exporter: swap pusher for exporter

### PR DESCRIPTION
**Pull request by ldelossa**
to provide consistent naming across the code base, deprecate pusher in
favor of exporter naming convention.

Signed-off-by: ldelossa <ldelossa@redhat.com>

### CODE REVIEW

**Comment by ldelossa** [comment]
A rebase of: #1486

**Comment by ldelossa** [comment]
@MrAlias - I closed the other PR, rebased master by accident instead of main. This is the same pr as 1486 be opened against latest main branch.

**Review by Aneurysm9 — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -61,6 +61,7 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
       "otel/sdk/trace".ReadOnlySpan
       "otel/sdk/trace".ReadWriteSpan
 ```
+- Swap "pusher" naming convention to "exporter" for consistent naming across codebase (#1656).
```
Comment (MrAlias) [inline CHANGELOG.md:64]: Can we update this to provide useful info to any user of the project?  Being explicit with the exported things that they will need to pay attention to. E.g.


```suggestion
- `"go.opentelemetry.io/sdk/metric/controller.basic"WithPusher` is replaced with `WithExporter` to provide consistent naming across project. (#1656)
```

Also, this needs to be moved to the `Unreleased` section.

**Review by MrAlias — CHANGES_REQUESTED** [review]
LGTM overall, just need to update the CHANGELOG

File: CHANGELOG.md (hunk 1 above)
Comment (ldelossa) [reply CHANGELOG.md:64]: definitely, makes sense.

**Review by MrAlias — APPROVED** [review]

**Pull request merged**